src/lib/discovery.ts

```diff
@@ -171,67 +180,68 @@ Discovery.prototype.log = function() {
 Discovery.prototype.discoverAllAPIs = function(discoveryUrl, callback) {
   const self = this;
   const headers = this.options.includePrivate ? {} : {'X-User-Ip': '0.0.0.0'};
-  transporter.request({uri: discoveryUrl, headers}, (err, resp) => {
-    if (err) {
-      return handleError(err, callback);
-    }
-
-    async.parallel(
-        resp.items.map(api => {
-          return (cb) => {
-            self.discoverAPI(api.discoveryRestUrl, (e, newApi) => {
+  transporter.request<DiscoverAPIsResponse>(
+      {url: discoveryUrl, headers}, (err, resp) => {
```

Comment:
nit: is it possible to use the promise version here?

Reply:
fixed!